### DISCUSSION TITLE
Make particles bounce on edge after being repulsed?

### DISCUSSION THREAD

**Discussion by hhuiminc** [matteobruni/tsparticles#4247] [Q&A] [answered]
Hi, I apologize if something similar has been answered - I have tried searching for solutions but I'm either not searching the right terms or no one has asked this. Also, this is my first time asking anything on github, forgive me if I'm not clear or if I leave any info out! I'm new to tsParticles too.

So I know that the `outModes` default to `bounce`, which works when the particles travel to the edge by itself. The question I have would be: How would I make it do the same thing when it is repulsed?

Right now, when my mouse brings them to the edge, they just go straight out, they don't bounce against the edge like when they hit the edge by themselves.

Another way to look at this would be: Is it possible to restrict particles from going outside the edge? Is there something I'm missing? Thanks in advance!

**Comment by matteobruni** [comment, marked as answer]
> Hi, I apologize if something similar has been answered - I have tried searching for solutions but I'm either not searching the right terms or no one has asked this. Also, this is my first time asking anything on github, forgive me if I'm not clear or if I leave any info out! I'm new to tsParticles too.
> 
> So I know that the `outModes` default to `bounce`, which works when the particles travel to the edge by itself. The question I have would be: How would I make it do the same thing when it is repulsed?
> 
> Right now, when my mouse brings them to the edge, they just go straight out, they don't bounce against the edge like when they hit the edge by themselves.
> 
> Another way to look at this would be: Is it possible to restrict particles from going outside the edge? Is there something I'm missing? Thanks in advance!

I try to answer to all questions:

First the default `outModes` is `out` and not `bounce`, which makes particles reappear from the other side.
To make particles bounce in a radius around the mouse you need to use the `bounce` mode in the `onHover` event, the repulse effect is slower, maybe you need to combine them a bit, the `onHover` mode can be an array as well (I haven't tested them together).

Last one I didn't understand the question, maybe you didn't set `outModes` to `bounce`, the `bounce` prevents particles from going outside the edges

**Reply by hhuiminc** [reply]
Thank you for the quick reply!

> First the default `outModes` is `out` and not `bounce`

Oh, oops. The code I have already had it set to `bounce`, so I thought it was the default.

> the `bounce` prevents particles from going outside the edges

Yep, this works when I let them roam and automatically hit the edge.

The question I was trying to ask previously was about the `repulse` pushing the particles out of the edges, not about the particles bouncing off the mouse radius (sorry if I was unclear!). 

So now, if I leave the particles to roam by themselves, they do bounce off edges, but when my mouse repulses them towards the edges, they don't bounce, they just go out (even though my `outModes` is already set to `bounce`).

This phrasing might help:
How do I make the particles bounce off edges when the mouse is repulsing them to the edges?

**Reply by matteobruni** [reply]
I think this could be a bug, I need to do some tests, particles shouldn't go out of edges with outmodes bounce

**Reply by hhuiminc** [reply]
Alright, thank you!
